main.tf

```diff
@@ -707,3 +712,25 @@ resource "aws_s3_bucket_ownership_controls" "this" {
     aws_s3_bucket.this
   ]
 }
+
+resource "aws_s3_bucket_intelligent_tiering_configuration" "tiering_conf" {
+  for_each = local.intelligent_tiering_config
```

Comment:
Add new line after for_each

Reply:
Such `for_each` should respect the value of `local.create_bucket` as in other resources.